### PR TITLE
Yeetcode

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- main action being done in this PR is removing `is_clickhouse_enabled()` and `PRIMARY_DB` + adjusting all side effects from doing so
- docker files also had to be updated as the ones that target postgres only deployments won't work anymore

*If this affects the frontend, include screenshots.*  

## How did you test this code?

*Please describe.*
Current tests should cover everything

### CODE REVIEW

**Comment by EDsCODE** [comment]
TODO:

- [x] restore e2e ci logging

**Comment by EDsCODE** [comment]
Going to want to run web and worker outside of docker so it doesn't build grpc everytime :/

**Comment by EDsCODE** [comment]
#7705 resolves the CI for e2e tests

**Comment by EDsCODE** [comment]
Cloud tests should work when merged. Currently using the docker-compose.dev on master which has been updated

**Comment by macobo** [comment]
Re cloud tests: You can test by changing the branch name from master it checks out temporarily

File: .deepsource.toml [hunk 1]
```diff
@@ -1,26 +0,0 @@
-version = 1
```
Comment (macobo) [inline .deepsource.toml:1]: What was this ile? Not sure it should be in this diff.

File: bin/docker-migrate [hunk 2]
```diff
@@ -1,4 +0,0 @@
-#!/bin/bash
```
Comment (macobo) [inline bin/docker-migrate:1]: Q: Why drop this here? Seems like an unrelated refactoring, bloating an already big migration?

File: .github/workflows/foss-sync.yml [hunk 3]
```diff
@@ -1,10 +1,10 @@
 name: Sync PostHog FOSS
 
-on:
-    push:
-        branches:
-            - master
-            - main
+# on:
```
Comment (macobo) [inline .github/workflows/foss-sync.yml:3]: Let's add a TODO here outlining what's going on or create an issue. This will get lost otherwise among other changes.

Also worth adding a banner at the top of posthog-foss.

File: .github/workflows/foss-release-image-publish.yml [hunk 4]
```diff
@@ -3,10 +3,10 @@
 
 name: Docker FOSS release image
 
-on:
-    push:
-        tags:
-            - '*.**'
+# on:
```
Comment (macobo) [inline .github/workflows/foss-release-image-publish.yml:6]: Same as below

File: .github/workflows/e2e.yml [hunk 5]
```diff
@@ -124,10 +121,9 @@ jobs:
             - name: Boot PostHog
               run: |
                   python manage.py collectstatic --noinput &
-                  ./bin/docker-migrate &
+                  ./bin/migrate &
```
Comment (macobo) [inline .github/workflows/e2e.yml:124]: This will not work. See `bin/check_kafka_clickhouse_up` call just below this.

File: .github/workflows/ci-plugin-server.yml [hunk 6]
```diff
@@ -36,146 +36,6 @@ jobs:
             - name: Lint with ESLint
               run: yarn lint
 
-    tests-postgres-1:
```
Comment (macobo) [inline .github/workflows/ci-plugin-server.yml:39]: I don't think this is safe to drop. This will make some legit tests not run I believe

We shouldn't include this in this PR and instead handle in a separate, plugin-server specific PR.

File: posthog/utils.py [hunk 7]
```diff
@@ -595,22 +595,16 @@ def queryset_to_named_query(qs: QuerySet, prepend: str = "") -> Tuple[str, dict]
...
 
 
-def is_clickhouse_enabled() -> bool:
-    return settings.EE_AVAILABLE and settings.PRIMARY_DB == AnalyticsDBMS.CLICKHOUSE
-
-
 def get_instance_realm() -> str:
     """
-    Returns the realm for the current instance. `cloud` or 'demo' or `hosted` or `hosted-clickhouse`.
+    Returns the realm for the current instance. `cloud` or 'demo' or `hosted-clickhouse`.
```
Comment (macobo) [inline posthog/utils.py:600]:
```suggestion
    Returns the realm for the current instance. `cloud` or 'demo' or `hosted-clickhouse`.
    
    Historically this would also have returned `hosted` for hosted postgresql based installations
```

Note this loses important historic context for someone looking through what property values exist.

File: posthog/tasks/status_report.py [hunk 8]
```diff
@@ -215,9 +193,6 @@ def fetch_sql(sql_: str, params: Tuple[Any, ...]) -> List[Any]:
 
 
 def get_instance_licenses() -> List[str]:
-    if settings.EE_AVAILABLE:
```
Comment (macobo) [inline posthog/tasks/status_report.py:218]: Sanity check question for @timgl - is EE_AVAILABLE env var safe to drop?

**Review by macobo — COMMENTED** [review]
Looked through it all, commented where I had concerns

The change itself is sane, but higher-risk than I'd like. I think we were over-eager changing some of the dockerfile plumbing logic and removing some code (having dead code is OK).

That said - get this in. If comments solved (including running tests w/ cloud repo) you have my :+1: to merge ASAP to avoid nasty conflicts.

File: posthog/tasks/status_report.py (hunk 8 above)
Comment (timgl) [reply posthog/tasks/status_report.py:218]: I don't see why not. My only worry is what happens if someone who is on postgres tries to upgrade? Is there a way for us to warn them? I guess not

**Review by Twixes — CHANGES_REQUESTED** [review]
+1 to eliminating `is_clickhouse_enabled()`, but it's not correct to remove `EE_AVAILABLE` here. We'll still have "enterprise edition", it's just that ClickHouse won't be a part of it.

File: posthog/settings/data_stores.py [hunk 9]
```diff
@@ -105,12 +105,7 @@
 KAFKA_HOSTS = ",".join(KAFKA_HOSTS_LIST)
 KAFKA_BASE64_KEYS = get_from_env("KAFKA_BASE64_KEYS", False, type_cast=str_to_bool)
 
-_primary_db = os.getenv("PRIMARY_DB", "postgres")
-PRIMARY_DB: AnalyticsDBMS
-try:
-    PRIMARY_DB = AnalyticsDBMS(_primary_db)
-except ValueError:
-    PRIMARY_DB = AnalyticsDBMS.POSTGRES
+PRIMARY_DB = AnalyticsDBMS.CLICKHOUSE
```
Comment (Twixes) [inline posthog/settings/data_stores.py:108]: What's the plan for this now that it's hardcoded and branches for the alternative value have been pruned all over?

File: posthog/settings/web.py [hunk 10]
```diff
@@ -68,10 +67,9 @@
     MIDDLEWARE.append("django_statsd.middleware.StatsdMiddlewareTimer")
 
 # Append Enterprise Edition as an app if available
-if EE_AVAILABLE:
```
[the remark this answers is not in the document]
Comment (EDsCODE) [reply posthog/settings/web.py:71]: I thought EE_AVAILABLE is meant to check if ee code is available not necessarily if a license exists. If we enable clickhouse by default in this PR without moving analytics code around, we're also going to need EE code always available. Paid features still have to be locked by a license check

[the remark this answers is not in the document]
Comment (Twixes) [reply posthog/settings/web.py:71]: That's true practically, but not from a licensing perspective. The `ee/` directory is the only one with a proprietary license, so we will definitely need to move ClickHouse code out of it for CH to actually be FOSS.

[the remark this answers is not in the document]
Comment (EDsCODE) [reply posthog/settings/web.py:71]: Ohh yeah for FOSS purposes definitely. That's why I'm trying to stop the FOSS syncing for a brief period until we move that code otherwise this PR would turn into too many lines more than it already is

File: bin/docker-migrate (hunk 2 above)
Comment (EDsCODE) [reply bin/docker-migrate:1]: point was to remove a never used in isolatoin command because you'd always want to migrate_clickhouse if we're defaulting to that support

Comment (EDsCODE) [reply bin/docker-migrate:1]: brought it back for now

File: .github/workflows/foss-sync.yml (hunk 3 above)
Comment (EDsCODE) [reply .github/workflows/foss-sync.yml:4]: done

File: posthog/settings/data_stores.py (hunk 9 above)
Comment (EDsCODE) [reply posthog/settings/data_stores.py:108]: Start purging this too. I opted not to in this PR because it's a bit involved also. Will do as follow up

**Comment by EDsCODE** [comment]
Can i get a look on these plugin server postgres jest tests

**Comment by EDsCODE** [comment]
@yakkomajuri Could you take a look at these postgres plugin tests? Context: I added a clickhouse dependency to the workflow since now we'll always require clickhouse when running the stack. The rest of the PR was around removing is_clickhouse_enabled. Can't figure out how/why those tests are failing now

File: .github/workflows/ci-backend.yml [hunk 11]
```diff
@@ -212,23 +212,13 @@ jobs:
               if: ${{ !matrix.foss }}
               run: bin/check_kafka_clickhouse_up
 
-            - name: Run FOSS tests
```
[the remark this answers is not in the document]
Comment (EDsCODE) [reply .github/workflows/ci-backend.yml:215]: correct, i changed it to just not rm ee folder

**Comment by yakkomajuri** [comment]
@EDsCODE essentially `KAFKA_ENABLED: true` messes up the Postgres tests. Better to leave that as is for now and revisit plugin server tests and all in another PR.

There's still one failing test but that feels like a flake.

**Comment by EDsCODE** [comment]
Update: Need to clean up FOSS tests. Removing the clickhouse_enabled flag changes the behavior of various foss tests (TestCapture patches are no longer needed but the payload needs to be updated, tests that have postgres factory functions but test a method that now uses only clickhouse implementation fail)

File: posthog/api/test/test_retention.py [hunk 12]
```diff
@@ -89,95 +81,3 @@ def get_retention(
             "properties": json.dumps(properties or []),
         },
     )
-
-
-@pytest.mark.django_db
-@freeze_time("2021-08-03")
```
Comment (EDsCODE) [inline posthog/api/test/test_retention.py:95]: PG backed test
